### PR TITLE
Port pipeline to DSL2

### PR DESCRIPTION
**Pull request by christopher-mohr**
This
- ports the `hlatyping` pipeline to DSL2
- uses existing nf-core modules
- merges nf-core template updates until `2.6`
- switches to samplesheet input 
- adds functionality for using `fastq` and `bam` files in the same run.

We still need a better solution for handling the HLA reference. Currently it is only possible to either process `dna` or `rna` data at the same time due to the HLA reference that has to be provided. My suggestion would be to choose the `.fasta` automatically based on the specified `seq_type` in the sample sheet AND allow users to specify custom `.fasta` files in the sample sheet (this might need then an additional process because of the `alleles.h5` file). 

In order to allow the use of different HLA references in the same run we also need to update the `yara_index` module and at least add the `meta` map to the process I guess. @apeltzer what do you think?

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Documentation in `docs` is updated
- [x] If necessary, also make a PR on the [nf-core/hlatyping branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/hlatyping)

### CODE REVIEW

**Comment by apeltzer** [comment]
Nice :) 
Yes, please work on an updated yara module - we've worked on that one when modules were started so some stuff wasn't fully clear back then on how to work with it. So an update of the module would be necessary I fear. 

Once you updated the module, I can review this here in detail

**Comment by apeltzer** [comment]
Please ping me on the yara module then @christopher-mohr

**Comment by christopher-mohr** [comment]
I updated the `yara` modules to the latest version and removed the `FASTA` parameter. The HLA references are now automatically set based on the given sequencing type. Not sure if there is an easier/ more elegant way for making sure that the reads are matched with the right index.

In addition I would add the possibility to provide a custom `FASTA` (for a specific sample) in the sample sheet in a separate PR.

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 FROM nfcore/base:1.10.2
 LABEL authors="Christopher Mohr, Alexander Peltzer, Sven Fillinger" \
-      description="Docker image containing all software requirements for the nf-core/hlatyping pipeline"
```
Comment (apeltzer) [inline Dockerfile:3]: We should ditch that entirely - all modules have their own now, so can just remove the dockerfile entirely.

File: README.md [hunk 2]
```diff
@@ -1,75 +1,87 @@
...
 
-The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It comes with docker containers making installation trivial and results highly reproducible.
+The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It uses Docker/Singularity containers making installation trivial and results highly reproducible. The [Nextflow DSL2](https://www.nextflow.io/docs/latest/dsl2.html) implementation of this pipeline uses one container per process which makes it much easier to maintain and update software dependencies. Where possible, these processes have been submitted to and installed from [nf-core/modules](https://github.com/nf-core/modules) in order to make them available to all nf-core pipelines, and to everyone within the Nextflow community!
 
-## Quick Start
+<!-- TODO nf-core: Add full-sized test dataset and amend the paragraph below if applicable -->
```
Comment (apeltzer) [inline README.md:20]: Remove please

File: README.md [hunk 3]
```diff
@@ -1,75 +1,87 @@
...
 
-Creates a config file from the command line arguments, which is then passed to OptiType. In parallel, the reads are extracted from the bam file and mapped again against the HLA reference sequence on chromosome 6. OptiType is then used for the HLA typing.
+## Credits
 
-![DAG with `.bam` file](docs/images/hlatyping_dag_bam.png)
+nf-core/hlatyping was originally written by [Christopher Mohr](https://github.com/christopher-mohr) from [Medical Data Integration Center](https://www.medizin.uni-tuebingen.de/de/das-klinikum/einrichtungen/institute/informationstechnologie-und-medizininformatik/medic) and [Quantitative Biology Center](https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qbic/), [Alexander Peltzer](https://github.com/apeltzer) from [Boeheringer Ingelheim](https://www.boehringer-ingelheim.de), and [Sven Fillinger](https://github.com/sven1103) from [Quantitative Biology Center](https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qbic/).
```
Comment (apeltzer) [inline README.md:64]:
```suggestion
nf-core/hlatyping was originally written by [Christopher Mohr](https://github.com/christopher-mohr) from [Medical Data Integration Center](https://www.medizin.uni-tuebingen.de/de/das-klinikum/einrichtungen/institute/informationstechnologie-und-medizininformatik/medic) and [Quantitative Biology Center](https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qbic/), [Alexander Peltzer](https://github.com/apeltzer) from [Boehringer Ingelheim](https://www.boehringer-ingelheim.de), and [Sven Fillinger](https://github.com/sven1103) from [Quantitative Biology Center](https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qbic/).
```

File: README.md [hunk 4]
```diff
@@ -1,75 +1,87 @@
...
 
-## Credits
+We thank the following people for their extensive assistance in the development of this pipeline:
 
-nf-core/hlatyping was originally written by [Christopher Mohr](https://github.com/christopher-mohr) from [Institute for Translational Bioinformatics](https://kohlbacherlab.org/team_tbi/) and [Quantitative Biology Center](https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qbic/),  [Alexander Peltzer](https://github.com/apeltzer) from [Böhringer Ingelheim](https://www.boehringer-ingelheim.de), and [Sven Fillinger](https://github.com/sven1103) from [Quantitative Biology Center](https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qbic/).
+<!-- TODO nf-core: If applicable, make list of people who have also contributed -->
```
Comment (apeltzer) [inline README.md:68]:
```suggestion
```

Or do we have other contributors who we should list here?

File: README.md [hunk 5]
```diff
@@ -1,75 +1,87 @@
...
 ## Contributions and Support
 
 If you would like to contribute to this pipeline, please see the [contributing guidelines](.github/CONTRIBUTING.md).
 
 For further information or help, don't hesitate to get in touch on the [Slack `#hlatyping` channel](https://nfcore.slack.com/channels/hlatyping) (you can join with [this invite](https://nf-co.re/join/slack)).
 
-## Citation
+## Citations
+
+<!-- TODO nf-core: Add citation for pipeline after first release. Uncomment lines below and update Zenodo doi and badge at the top of this file. -->
```
Comment (apeltzer) [inline README.md:78]:
```suggestion
```

File: assets/methods_description_template.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+id: "nf-core-hlatyping-methods-description"
+description: "Suggested text and references to use when describing pipeline usage within the methods section of a publication."
+section_name: "nf-core/hlatyping Methods Description"
+section_href: "https://github.com/nf-core/hlatyping"
+plot_type: "html"
+## TODO nf-core: Update the HTML below to your prefered methods description, e.g. add publication citation for this pipeline
```
Comment (apeltzer) [inline assets/methods_description_template.yml:6]:
```suggestion
```

File: docs/usage.md [hunk 7]
```diff
@@ -1,23 +1,85 @@
 # nf-core/hlatyping: Usage
 
+## :warning: Please read this documentation on the nf-core website: [https://nf-co.re/hlatyping/usage](https://nf-co.re/hlatyping/usage)
+
+> _Documentation of pipeline parameters is generated automatically from the pipeline schema and can no longer be found in markdown files._
+
 ## Introduction
 
+The hlatyping pipeline can currently deal with two input formats: `.fastq{.gz}` or `.bam`, not both at the same time however. If the input file type is `bam`, than the pipeline extracts all reads from it and performs an mapping additional step with the `yara` mapper against the HLA reference sequence. Indices are provided in the `./data` directory of this repository. Optitype uses [razers3](https://github.com/seqan/seqan/tree/master/apps/razers3), which is very memory consuming. In order to avoid memory issues during pipeline execution, we reduce the mapping information on the relevant HLA regions on chromosome 6.
```
Comment (apeltzer) [inline docs/usage.md:9]: Is this still true? I thought the new version can handle both ...?

Comment (apeltzer) [reply docs/usage.md:9]: Indices are not in .data anymore - please remove that part :-)

**Review by apeltzer — APPROVED** [review]
Looks good to have this on DSL2 now. Some minor comments. I will also run once this is on dev to see if docs are matching expectations 👍🏻

File: README.md (hunk 4 above)
Comment (christopher-mohr) [reply README.md:68]: Let's leave it in there for now, so we don't forget it.

**Pull request merged**